src/ReasonReactRouter.re

```diff
@@ -67,50 +67,68 @@ let arrayToList = a => {
 /* sigh URLSearchParams doesn't work on IE11, edge16, etc. */
 /* actually you know what, not gonna provide search for now. It's a mess.
    We'll let users roll their own solution/data structure for now */
-let path = () =>
-  switch ([%external window]) {
-  | None => []
-  | Some((window: Dom.window)) =>
-    switch (window |> location |> pathname) {
-    | ""
-    | "/" => []
-    | raw =>
-      /* remove the preceeding /, which every pathname seems to have */
-      let raw = Js.String.sliceToEnd(~from=1, raw);
-      /* remove the trailing /, which some pathnames might have. Ugh */
-      let raw =
-        switch (Js.String.get(raw, Js.String.length(raw) - 1)) {
-        | "/" => Js.String.slice(~from=0, ~to_=-1, raw)
-        | _ => raw
-        };
-      raw |> Js.String.split("/") |> arrayToList;
-    }
+let hsMatcher = [%re "/(#[^?&]+)?([?&].*)?$/"];
+let pathMatcher = [%re "/(#|\\?).+$/"];
+let pathParse = str =>
+  switch (str) {
+  | ""
+  | "/" => []
+  | raw =>
+    /* remove the preceeding /, which every pathname seems to have */
+    let raw = Js.String.sliceToEnd(~from=1, raw);
+    /* remove the trailing /, which some pathnames might have. Ugh */
+    let raw =
+      switch (Js.String.get(raw, Js.String.length(raw) - 1)) {
+      | "/" => Js.String.slice(~from=0, ~to_=-1, raw)
+      | _ => raw
+      };
+    raw
+    |> Js.String.replaceByRe(pathMatcher, "")
+    |> Js.String.split("/")
+    |> Js.Array.filter(item => String.length(item) != 0)
+    |> arrayToList;
   };
-let hash = () =>
-  switch ([%external window]) {
-  | None => ""
-  | Some((window: Dom.window)) =>
-    switch (window |> location |> hash) {
-    | ""
-    | "#" => ""
-    | raw =>
-      /* remove the preceeding #, which every hash seems to have.
-         Why is this even included in location.hash?? */
-      raw |> Js.String.sliceToEnd(~from=1)
+
+let path = (~serverUrlString=None, ()) =>
```

Comment:
This pattern is a bit odd. Just go for ~serverUrlString=?.

Reply:
Oh sure, I used that pattern to not have to check inside of `url()` but yeah I switch it to `?`